src/sentry/static/sentry/app/main.jsx

```diff
@@ -7,17 +7,24 @@ import routes from 'app/routes';
 import {loadPreferencesState} from 'app/actionCreators/preferences';
 import * as tracing from 'app/utils/tracing';
 
+function updateTracingData () {
+  tracing.setTransactionId()
+  tracing.setSpanId()
+  tracing.setCurrentRoute(browserHistory.getCurrentLocation().pathname)
```

Comment:
This is fine as a default but it will be really noisy and has the chance to leak PII if we were to do this for customers.

Reply:
I’m fact I would challenge this is not useful at all to us. We really need a normalized transaction name either via component name or route name. I’d rether we do that than make this partial change otherwise we still won’t reliably be able to say “top 10 transactions with errors”.

Reply:
We could move tracing to `app/views/app` and use https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/utils/getRouteStringFromRoutes.jsx to get a more normalized route.

Reply:
I tried this @billyvg, but `App` component doesn't receive `this.state.route` updates when child component (page) changes.

> component name or route name

Afair it's not possible without manually keeping track of this data. Unless we are not talking about route name being `something/:org/:id` format.

The problem here is that `Main` is the only place that can modify Sentry's config on every route change, no matter error happened or not.

Reply:
Is it possibly to hook react-router events and get the normalized route from that?

Reply:
@kamilogorek It is possible, we can move the browserHistory listener into `app/views/app`, and use the `withRouter` HoC on `<App>` - App component will then have a `this.props.routes` array which you can pass to `getRouteStringFromRoutes()`